Dryad Digital Repository.js

```diff
@@ -0,0 +1,686 @@
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+var doiUrlBase      = 'http://dx.doi.org/';
+var packageXpath    = '//meta[@name="DCTERMS.isPartOf"]/@content';
+var multiplesXpath  = '//li[contains(@class,"ds-artifact-item")]/div/a';
+var doiXpath        = '//meta[@name="DC.identifier"][starts-with(@content,"doi:10.5061/dryad.")]/@content';
+var dcTypeXpath     = '//meta[@name="DC.type"][1]/@content';
+var discoverPage    = /^https?:\/\/(www\.)?datadryad\.org\/discover/;
+var urlToDoi        = /^.*?doi:/;
+
+function detectWeb(doc, url) {
+	// Dryad search page
+	if (url.match(discoverPage)) {
+		return 'multiple';
+	} else {
+		var result = ZU.xpathText(doc,dcTypeXpath);
+        // Dryad data package
+        if (result === 'Article') {
+            return 'journalArticle';
+        // Dryad data file
+        } else if (result === 'Dataset') {
+            //return 'dataset';
+            return 'journalArticle';
+        }
+	}
+	return false;
+}
+
+function doWeb(doc, url) {
+	var itemType = detectWeb(doc, url);
+	if (itemType === 'journalArticle') {
+		var translator = Zotero.loadTranslator('import');
+		// use the Embedded Metadata translator
+		translator.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48');
+		// value for page's possible DCTERMS.isPartOf <meta> tag,
+		// minus the initial 'doi:'
+		var package_doi = ZU.xpathText(doc, packageXpath);
+		if (package_doi) package_doi = package_doi.substring('doi:'.length);
+		// note: this handler function is called for a data package
+		// or a data file. If called for a data package as part of setting
+		// the seeAlso value, then the 'doc' variable is still aliasing
+		// the data file's DOM
+		translator.setHandler('itemDone', function (obj, item) {
+			var itemDoi   = item.url.replace(urlToDoi, '');
+			item.DOI      = itemDoi;
+			item.itemId   = itemDoi;      // internal value for seeAlso relation
+			item.url      = '';           // the doi value is to be used in lieu of the url
+			item.itemType = itemType;
+			// Add a .seeAlso value if this is a page for a single data file,
+			// rather than for a data package collection.
+			// This is a data file if there is a DCTERMS.isPartOf property.
+			item.seeAlso = (package_doi && package_doi !== itemDoi)
+                ? [doiUrlBase + package_doi]    // data file 
+                : [];                           // data package
+			// signal a dataset until supported per
+			// https://www.zotero.org/support/dev/translators/datasets
+			item.extra = '{:itemType: dataset}';
+			item.archive = 'Dryad Digital Repository';
+			item.attachments = [];
+			item.shortTitle = '';
+			item.complete();
+		}); // translator.setHandler()
+		translator.getTranslatorObject(function (obj) {
+			obj.doWeb(doc, url);
+		});
+		if (package_doi) {
+			ZU.processDocuments(doiUrlBase + package_doi, function (d) {
+				doWeb(d, d.location.href);
```

Comment:
I updated this call to processDocuments, but I am seeing a failure when processDocuments is called with an array of urls rather than a single url. When called with an array, the callback is executed only once, so I have left the other processDocuments occurrence with an anonymous function.